src/cdk/testing/protractor/protractor-element.ts

```diff
@@ -6,248 +6,12 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {
-  _getTextWithExcludedElements,
-  ElementDimensions,
-  ModifierKeys,
-  TestElement,
-  TestKey,
-  TextOptions,
-  EventData,
-} from '@angular/cdk/testing';
-import {browser, Button, by, ElementFinder, Key} from 'protractor';
-
-/** Maps the `TestKey` constants to Protractor's `Key` constants. */
-const keyMap = {
-  [TestKey.BACKSPACE]: Key.BACK_SPACE,
-  [TestKey.TAB]: Key.TAB,
-  [TestKey.ENTER]: Key.ENTER,
-  [TestKey.SHIFT]: Key.SHIFT,
-  [TestKey.CONTROL]: Key.CONTROL,
-  [TestKey.ALT]: Key.ALT,
-  [TestKey.ESCAPE]: Key.ESCAPE,
-  [TestKey.PAGE_UP]: Key.PAGE_UP,
-  [TestKey.PAGE_DOWN]: Key.PAGE_DOWN,
-  [TestKey.END]: Key.END,
-  [TestKey.HOME]: Key.HOME,
-  [TestKey.LEFT_ARROW]: Key.ARROW_LEFT,
-  [TestKey.UP_ARROW]: Key.ARROW_UP,
-  [TestKey.RIGHT_ARROW]: Key.ARROW_RIGHT,
-  [TestKey.DOWN_ARROW]: Key.ARROW_DOWN,
-  [TestKey.INSERT]: Key.INSERT,
-  [TestKey.DELETE]: Key.DELETE,
-  [TestKey.F1]: Key.F1,
-  [TestKey.F2]: Key.F2,
-  [TestKey.F3]: Key.F3,
-  [TestKey.F4]: Key.F4,
-  [TestKey.F5]: Key.F5,
-  [TestKey.F6]: Key.F6,
-  [TestKey.F7]: Key.F7,
-  [TestKey.F8]: Key.F8,
-  [TestKey.F9]: Key.F9,
-  [TestKey.F10]: Key.F10,
-  [TestKey.F11]: Key.F11,
-  [TestKey.F12]: Key.F12,
-  [TestKey.META]: Key.META
-};
-
-/** Converts a `ModifierKeys` object to a list of Protractor `Key`s. */
-function toProtractorModifierKeys(modifiers: ModifierKeys): string[] {
-  const result: string[] = [];
-  if (modifiers.control) {
-    result.push(Key.CONTROL);
-  }
-  if (modifiers.alt) {
-    result.push(Key.ALT);
-  }
-  if (modifiers.shift) {
-    result.push(Key.SHIFT);
-  }
-  if (modifiers.meta) {
-    result.push(Key.META);
-  }
-  return result;
-}
+import {WebDriverElementBase} from '@angular/cdk/testing/webdriver';
+import {ElementFinder} from 'protractor';
 
 /** A `TestElement` implementation for Protractor. */
-export class ProtractorElement implements TestElement {
-  constructor(readonly element: ElementFinder) {}
-
-  async blur(): Promise<void> {
-    return browser.executeScript('arguments[0].blur()', this.element);
-  }
-
-  async clear(): Promise<void> {
-    return this.element.clear();
-  }
-
-  async click(...args: [ModifierKeys?] | ['center', ModifierKeys?] |
-    [number, number, ModifierKeys?]): Promise<void> {
-    await this._dispatchClickEventSequence(args, Button.LEFT);
-  }
-
-  async rightClick(...args: [ModifierKeys?] | ['center', ModifierKeys?] |
-    [number, number, ModifierKeys?]): Promise<void> {
-    await this._dispatchClickEventSequence(args, Button.RIGHT);
-  }
-
-  async focus(): Promise<void> {
-    return browser.executeScript('arguments[0].focus()', this.element);
+export class ProtractorElement extends WebDriverElementBase {
+  constructor(readonly element: ElementFinder) {
+    super(() => element.getWebElement(), async () => {});
```

Comment:
It looks like the e2e tests fail because there is no stabilize function, and we no longer go through the Protractor patched functions (that rely on the `blocking-proxy`). Should we just use the stabilizers as we do for plain webdriver?

Reply:
Alternatively, we might be able to just use the Protractor `browser.waitForAngular` function?
https://github.com/angular/protractor/blob/df3dd6400e3a4792ff9f7a58849d2a5eed6e1483/lib/browser.ts#L620

Reply:
Hmm, oddly passing the `waitForAngular` function didn't seem to have any effect